src/game_inventory.cpp

```diff
@@ -1614,9 +1620,128 @@ class bionic_install_preset: public inventory_selector_preset
+            append_cell( [this]( const item_location & loc ) {
+                return get_failure_chance( loc );
+            }, _( "FAILURE CHANCE" ) );
+
+            append_cell( [this]( const item_location & loc ) {
+                return get_operation_duration( loc );
+            }, _( "OPERATION DURATION" ) );
+
+            append_cell( [this]( const item_location & loc ) {
+                return get_money_amount( loc );
+            }, _( "PRICE" ) );
+        }
+
+        bool is_shown( const item_location &loc ) const override {
+            return loc->is_bionic();
+        }
+
+        std::string get_denial( const item_location &loc ) const override {
+            const item *it = loc.get_item();
+            const itype *itemtype = it->type;
+            const bionic_id &bid = itemtype->bionic->id;
+
+            if( it->has_flag( "FILTHY" ) ) {
+                return _( "CBM is filthy." );
+            } else if( !it->has_flag( "STERILE" ) ) {
+                return _( "CBM is not sterile." );
+            } else if( pa.has_bionic( bid ) ) {
+                return _( "CBM already installed" );
+            } else if( bid->upgraded_bionic &&
+                       !pa.has_bionic( bid->upgraded_bionic ) &&
+                       it->is_upgrade() ) {
+                return _( "No base version installed" );
+            } else if( std::any_of( bid->available_upgrades.begin(),
+                                    bid->available_upgrades.end(),
+                                    std::bind( &player::has_bionic, &pa,
+                                               std::placeholders::_1 ) ) ) {
+                return _( "Superior version installed" );
+            } else if( pa.is_npc() && !bid->npc_usable ) {
+                return _( "CBM not compatible with patient" );
+            } else if( loc.get_item()->price( true ) * 2 > p.cash ) {
+                return format_money( loc.get_item()->price( true ) * 2 );
+            }
+
+            return std::string();
+        }
+
+    protected:
+        player &p;
+        player &pa;
+
+    private:
+        // Returns a formatted string of how long the operation will take.
+        std::string get_operation_duration( const item_location &loc ) {
+
+            const int difficulty = loc.get_item()->type->bionic->difficulty;
+
+            // 20 minutes per bionic difficulty.
+            int hours = difficulty / 3;
+            int minutes = ( difficulty % 3 ) * 20;
+            std::string minutes_string = to_string_clipped( time_duration::from_minutes( difficulty * 20 ) );
+
+            if( hours > 0 ) {
+                std::string hours_string = hours >= 2
+                                           ? string_format( _( "%i hours" ), hours )
+                                           : string_format( _( "%i hour" ), hours );
+
+                if( minutes > 0 ) {
+                    return string_format( _( "%s, %s" ), hours_string, minutes_string );
```

Comment:
If you want two time units (e.g. "2 hours, 20 minutes"), use `to_string( time_duration )`. See documentation of those to_string* functions in "calendar.h".